spec/integration/parser/compiler_spec.rb

```diff
@@ -522,4 +522,31 @@ class xport_test {
       expect(catalog).to have_resource("Notify[nbr2]").with_parameter(:message, 'overridden')
     end
   end
+
+
+  context "when converting catalog to resource" do
+    it "the same environment is used for compilation as for transformation to resource form" do
+        Puppet[:code] = <<-MANIFEST
+          notify { 'dummy':
+          }
+        MANIFEST
+
+        # Create an instance of a mocha state machine that checks that the environment is
+        # 'production'
+        in_production_state = Class.new do
+
+          def active?
+            Puppet.lookup(:current_environment).name == :production
+          end
+
+          def to_s
+            # displayed if the expectation about this state fails
+            "in production environment"
+          end
+        end.new
```

Comment:
if I understood the StateMachine::State class it is intended to describe a set of static states, and there are methods to set/get and transition between such states. Basically, our need is to return an instance of an object that responds to `active` - the same can be achieved through other less contorted means than the anonymous class I came up with (I thought I needed to implement additional methods) - OTOH, such an implementation will probably be as hard to read as the anonymous class. So - in short; no, I don't see how the standard StateMachine::State can be used - I did not see an apparent way of doing it at least.

I do not think it matters if we test the default env, or some other env. It starts with the *root* env in both cases. I contemplated adding an assertion in the test that the current_environment is not `production` before the test - is that good enough?